src/boolean3.cpp

```diff
@@ -88,35 +92,23 @@ struct CopyFaceEdges {
   }
 };
 
-SparseIndices Filter11(const Manifold::Impl &inP, const Manifold::Impl &inQ,
-                       const SparseIndices &p1q2, const SparseIndices &p2q1) {
-  ZoneScoped;
-  SparseIndices p1q1(3 * p1q2.size() + 3 * p2q1.size());
-  for_each_n(autoPolicy(p1q2.size(), 1e5), countAt(0_uz), p1q2.size(),
-             CopyFaceEdges<false>({p1q2, p1q1, inQ.halfedge_, 0_uz}));
-  for_each_n(autoPolicy(p2q1.size(), 1e5), countAt(0_uz), p2q1.size(),
-             CopyFaceEdges<true>({p2q1, p1q1, inP.halfedge_, p1q2.size()}));
-  p1q1.Unique();
-  return p1q1;
-}
-
 inline bool Shadows(double p, double q, double dir) {
   return p == q ? dir < 0 : p < q;
 }
 
 inline std::pair<int, vec2> Shadow01(
     const int p0, const int q1, VecView<const vec3> vertPosP,
     VecView<const vec3> vertPosQ, VecView<const Halfedge> halfedgeQ,
-    const double expandP, VecView<const vec3> normalP, const bool reverse) {
+    const double expandP, VecView<const vec3> normal, const bool reverse) {
```

Comment:
changed the naming because it is not really for `P`, but depends on `reverse`. When `reverse` is false, it is for `P`, but otherwise it is for `Q`.

Reply:
Okay, so this one is weird - I managed to confuse myself a bunch of times, which means my naming is terrible, and part of the problem is I added on this `expandP` thing late. At a global level, only P is expanded (or contracted if false), according to its own normals. The normals of Q are never used for anything. This is weird because this function reverses P and Q sometimes, but the normals are always the same. I *think* I've used them right, but a double-check would be great. As for naming... I don't know.